Fenced_Frames_Ads_Reporting.md

```diff
@@ -100,4 +100,50 @@ registerAdBeacon({
 });
 ```
 
+## Support for Attribution Reporting 
+### Goals
+*   While fenced frames still have unrestricted network access and FLEDGE supports event-level reporting, the solution below takes advantage of registerAdBeacon/reportEvent information flow to enable [registering attribution sources](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#registering-attribution-sources). [ARA attribution triggering](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#triggering-attribution) is unchanged for registered FLEDGE impressions.
+*   Improve the ergonomics of triggering ad beacons based on clicks.
+
+**As of writing this, the changes below are not yet implemented in Chrome.**
+
+### registerAdBeacon
+
+The worklet code will be able to register an event called “reserved.top_navigation” via registerAdBeacon. 
+
+```
+registerAdBeacon({
+ 'reserved.top_navigation': 'https://adtech.example/click?buyer_event_id=123',
+});
+```
+
+The new event, if registered, implies that an automatic beacon will be sent by the browser to the registered url when a top-level navigation is invoked from within the fenced frame. This will impact top-level navigation initiated from the fenced frame in the same tab (via [unfencedTop target](https://github.com/WICG/fenced-frame/blob/master/explainer/integration_with_web_platform.md#top-level-navigation)) or in a different tab. Note that this beacon is gated on a transient user activation. More details about the beacon are below.
+
+
+### reportEvent
+
+The beacons that are generated from reportEvent invocation or via the automatic reserved.top_navigation event will now be automatically eligible for attribution, i.e. they would be appended with `Attribution-Reporting-Eligible` request header. The beacon responses can then register sources as usual, as described [here](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#registering-attribution-sources).
+
+#### Redirects
+
+As mentioned in the explainer above, reportEvent beacons are POST requests and carry eventData in the request's body. The same will be true for automatic reserved.top_navigation requests. Note that for any server redirects of the initial request, the browser sends a GET request and does not include the initial request's body. For attribution registration flow, if the eventData needs to be used as part of the redirected request, it must be explicitly passed on as part of the redirect URL.
+
+### API to populate event data for reserved.top_navigation
+
+Since reserved.top\_navigation beacons are automatically generated by the browser, there needs to be some way for those beacons to be associated with and append event data, as it happens in reportEvent generated beacons. To acheive this, the following API can be invoked from within the fenced frame:
```

Comment:
```suggestion
Since the `reserved.top_navigation` beacons are automatically generated by the browser, there needs to be some way for those beacons to be associated with and include event data, as it happens in `reportEvent` generated beacons. To achieve this, a new `setReportEventDataForAutomaticBeacons` API can be invoked from within the fenced frame:
```